src/app/pipes/folder-view.pipe.ts

```diff
@@ -62,19 +64,35 @@ export class FolderViewPipe implements PipeTransform {
    */
   extractFourPreviewHashes(files: ImageElement[]): string {
     let hashes: string = '';
+    if (files.length > 4 && this.settingsButtons['randomizeFoldersScreenshots'].toggled) {
+      hashes = this.extractRandomPreviewHashes(files);
+    } else {
+      hashes = this.extractFirstFourPreviewHashes(files);
+    }
+    if (hashes.charAt(0) === ':') {
+      hashes = hashes.slice(1);
+    }
+    return hashes;
+  }
 
+  extractFirstFourPreviewHashes(files: ImageElement[]): string {
+    let hashes: string = '';
     for (let n = 0; n < 4; n++) {
       if (files[n]) {
         hashes += ':' + files[n].hash;
       } else {
         break;
       }
     }
+    return hashes;
+  }
 
-    if (hashes.charAt(0) === ':') {
-      hashes = hashes.slice(1);
+  extractRandomPreviewHashes(files: ImageElement[]): string {
+    let hashes: string = '';
+    for (let index = 0; index < 4; index++) {
+      const randomIndex = Math.floor(Math.random() * files.length);
+      hashes += ':' + files.splice(randomIndex, 1)[0].hash;
```

Comment:
🤔 `.splice` removes elements _in place_ [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) 🤔 

> The splice() method changes the contents of an array by removing or replacing existing elements and/or adding new elements in place.

I think this _may_ be fine here, but I'd like your thoughts and a comment on this 🤔

Reply:
I suspect it's fine because I think we're working with a temporary copy and not at all messing with the original `finalArray` 😅 -- and I tested the code -- it looks good and behaves well, so I think we're clear ... but still would appreciate a comment 👍 

ps - This file was a tense algorithm to write 😅

Reply:
Yeah,
So i got this thought myself when i wrote it. But than i got to the same conclusion that this is only a copy and this array is used in order to get the screenshots anyway.
It took me a while to think about how to do it. And this seems the most proficient way to get it done and get unique 4 hashes.